File: rustsem-ocaml/cl-src/.../tests/t4.m

#allocateInit(2,i32,valcl(1),valcl(2))
#allocateInit(2,i32,valcl(1),valcl(2),valcl(4))
#allocateInit(3,i32,valcl(1),valcl(2),valcl(3))


#allocStack

#allocStack

#allocStack

#write(#sl(0),#own(#blk(0)))
#write(#sl(1),#Shr(#sl(0)))    //#sl( 1 ) borrows #sl(0) here

#write(#sl(0),#own(#blk(1)))   //#sl( 0 ) being written here

#read(#dRef(#sl(1)))           //#sl( 1 ) being used here

File: rustsem-ocaml/rust_tests/rv/rduninit/t4.rs

struct A{a:i32}

fn main(){

   let a0 = A{a:0};
   let a1 = A{a:1};
   let a2 = A{a:2};
   let a3 = A{a:3};
   let a4 = A{a:4};

   let v = [a0,a1,a2,a3,a4];

   let t = v[2];

   println!("{}",(v[2]).a);

}